sqlglot/dialects/singlestore.py

```diff
@@ -1742,6 +1761,23 @@ def datatype_sql(self, expression: exp.DataType) -> str:
                     return f"DECIMAL({precision}, {scale[0]})"
                 else:
                     return f"DECIMAL({precision})"
+            if expression.is_type(exp.DataType.Type.VECTOR):
+                expressions = expression.expressions
+                if len(expressions) == 2:
+                    type_name = self.sql(expressions[0])
+                    if type_name == "TINYINT":
+                        type_name = "I8"
+                    elif type_name == "SMALLINT":
+                        type_name = "I16"
+                    elif type_name == "INT":
+                        type_name = "I32"
+                    elif type_name == "BIGINT":
+                        type_name = "I64"
+                    elif type_name == "FLOAT":
+                        type_name = "F32"
+                    elif type_name == "DOUBLE":
+                        type_name = "F64"
+                    return f"VECTOR({self.sql(expressions[1])}, {type_name})"
```

Comment:
Actually, why do we even need this? Is it just for transpilation? Given that you always map these types to their aliases at parse time..

Reply:
It is for transpilation. I'm trying to match Snowflake parsing/generation logic where the first expression is the type and the second is the dimension.